### PR TITLE
Support adding 'tags' to frontmatter to improve Search-ability of content

### PR DESCRIPTION
**Pull request by w8896699**
- integrate tags into nodes
-  tag can be indexed by search.

### CODE REVIEW

File: app-web/gatsby/onCreateNode.js [hunk 1]
```diff
@@ -261,6 +267,19 @@ module.exports = ({ node, actions, getNode, getNodes }) => {
...
       value: labels,
     });
+
+    createNodeField({
+      node,
+      name: 'tags',
+      value: tags,
+    });
+
+    createNodeField({
```
Comment (patricksimonian) [inline app-web/gatsby/onCreateNode.js]: should only create parentNode on github raw nodes.

File: app-web/src/hoc/withResourceQuery.js [hunk 2]
```diff
@@ -56,6 +56,7 @@ const withResourceQuery = WrappedComponent => () => props => (
             node {
               fields {
                 title
+                tags
```
Comment (patricksimonian) [inline app-web/src/hoc/withResourceQuery.js]: why do we need this field?

File: app-web/src/pages/index.js [hunk 3]
```diff
@@ -106,18 +106,18 @@ const getSearchResultTotal = results => {
...
   const resourcesSelector = selectResourcesGroupedByType();
   let resourcesToShow = [];
   if (!isNull(results) && results.length > 0) {
     //map the search index results to the resources. Its important to do it in this order,
     //since the index results are return in order based on relevance
     resourcesToShow = results.flatMap(result => {
-      return resources.filter(resource => result.id === resource.id);
-    });
+      return resources.filter(resource => result.id === resource.id); //this already remove duplicate, and have some potential issue as some resource will be filter out, same item has different id in resources and result.
+    }); // TODO, investigate in above two lines.
```
Comment (patricksimonian) [inline app-web/src/pages/index.js]: ??

Comment (patricksimonian) [reply app-web/src/pages/index.js]: i think we need to do some further investigtion on how ids are intersected between results and resources.

**Comment by patricksimonian** [comment]
I can create a PR into this one that will remove devhub siphon plugin's abliity to create github nodes?

**Comment by w8896699** [comment]
That works for me.

**Comment by patricksimonian** [comment]
this pr also needs to be rebased on master

File: app-web/src/pages/index.js [hunk 4]
```diff
@@ -85,14 +83,14 @@ const getTopicPreviews = (topics, searchResultsExist) => {
...
  */
-const getUniqueResources = resources => {
-  let events = resources.filter(resource => resource.fields.resourceType === RESOURCE_TYPES.EVENTS);
-  let allButEvents = resources.filter(
-    resource => resource.fields.resourceType !== RESOURCE_TYPES.EVENTS,
-  );
-  allButEvents = uniqBy(allButEvents, 'fields.title');
-  return allButEvents.concat(events);
-};
+// const getUniqueResources = resources => {
```
Comment (patricksimonian) [inline app-web/src/pages/index.js]: why . is this commented out?

File: app-web/src/pages/index.js [hunk 5]
```diff
@@ -106,20 +104,21 @@ const getSearchResultTotal = results => {
...
+
   if (!isNull(results) && results.length > 0) {
     //map the search index results to the resources. Its important to do it in this order,
     //since the index results are return in order based on relevance
     resourcesToShow = results.flatMap(result => {
       return resources.filter(resource => result.id === resource.id);
     });
-    //remove any duplicates
-    resourcesToShow = getUniqueResources(resourcesToShow);
+    //remove any duplicates//// give it a try, if no complanint, next person who read this line please remove this.
```
Comment (patricksimonian) [inline app-web/src/pages/index.js]: NOPE

**Pull request merged**